noxfile.py

```diff
@@ -173,13 +174,25 @@ def lint(session: nox.sessions.Session):
         A `nox.sessions.Session` object.
 
     """
+    import yaml
+
     # Pip install the session requirements.
     session.install("pre-commit")
-    # Execute the pre-commit linting tasks.
-    cmd = ["pre-commit", "run", "--all-files"]
-    hooks = ["black", "blacken-docs", "flake8", "isort"]
+
+    # Load the pre-commit configuration YAML file.
+    with open(".pre-commit-config.yaml", "r") as fi:
+        config = yaml.load(fi, Loader=yaml.FullLoader)
+
+    # Enumerate the pre-commit hooks.
+    hooks = [
+        Path(urlsplit(unquote(entry["repo"])).path).name
+        for entry in config["repos"]
+    ]
```

Comment:
It it safe to depend on the URL? Presumably pre-commit is actually attempting to match your `hook` argument below to an `id` value under `hooks`?

Reply:
TBH I'm not totally mad keen on parsing a URL, although it is pretty easy to do and I think stable enough for our needs.

I'll have a play and drill down to the `id`, but some repos provide more than one hook, but that's shouldn't be a blocker... let's see :+1:

Reply:
Yeah, I think we've tended towards a nice approach now... happy to go with that now?

Reply:
> but some repos provide more than one hook

Yeah, that's unfortunate, but I think it's a pre-commit API limitation rather than something we can solve with the right incantation. The final argument is [specifying a single hook id](https://pre-commit.com/#pre-commit-run), so we're already stuck running just one hook whether we extract the hook's id from a URL or from somewhere else.

Reply:
The `id` is guaranteed per hook, and `alias` is optional, so I think this is pretty robust, https://pre-commit.com/#pre-commit-configyaml---hooks